src/App.tsx

```diff
@@ -1,24 +1,187 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import {
+  deleteTodo, getTodos, postTodo, updateTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoFooter } from './components/TodoFooter';
+import { FilterType } from './types/FilterType';
+import { getFilteredTodos } from './utils/functions';
+import { ErrorType } from './types/ErrorType';
 
-const USER_ID = 0;
+const USER_ID = 7002;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [error, setError] = useState<ErrorType | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  const activeTodosQuantity = todos.filter((todo) => !todo.completed).length;
+  const completedTodosQuantity = todos.length - activeTodosQuantity;
+
+  const showErrorNotification = (errorType: ErrorType) => {
+    setError(errorType);
+    setTimeout(() => setError(null), 3000);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        showErrorNotification(ErrorType.LOAD);
+      });
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    return getFilteredTodos(todos, filterType);
+  }, [todos, filterType]);
+
+  const handleAddTodo = (title: string) => {
+    if (title.length === 0) {
+      showErrorNotification(ErrorType.EMPTY);
+    } else {
+      const newTodo = {
+        id: 0,
+        userId: USER_ID,
+        completed: false,
+        title,
+      };
+
+      setTempTodo(newTodo);
+      setError(null);
+
+      postTodo(newTodo)
+        .then((todo) => {
+          setTodos((prevTodos) => {
+            return [...prevTodos, todo];
+          });
+        })
+        .catch(() => {
+          showErrorNotification(ErrorType.ADD);
+        })
+        .finally(() => {
+          setTempTodo(null);
+        });
+    }
+  };
+
+  const handleDeleteTodo = useCallback(
+    (id: number) => {
+      setLoadingTodoIds((state) => [...state, id]);
+
+      return deleteTodo(id)
+        .then(() => {
+          setTodos((prevTodos) => {
+            return prevTodos.filter((todo) => todo.id !== id);
+          });
+        })
+        .catch(() => {
+          showErrorNotification(ErrorType.DELETE);
+        })
+        .finally(() => {
+          setTempTodo(null);
+          setLoadingTodoIds([]);
+        });
+    },
+    [deleteTodo],
+  );
+
+  const handleClearCompleted = () => {
+    todos.filter((todo) => todo.completed).forEach((todo) => {
+      handleDeleteTodo(todo.id)
+        .then(() => setTodos(todos.filter(({ completed }) => !completed)));
+    });
```

Comment:
O(2n) -> O(n) and avoid many re-renders

```suggestion
    todos.forEach((todo) => {
      if (todo.completed) {
        handleDeleteTodo(todo.id)
      }
    });
    
    setTodos(todos.filter(({ completed }) => !completed)
```